src/sentry/release_health/metrics_sessions_v2.py

```diff
@@ -312,13 +318,65 @@ def get_values(
             ]
 
 
+class _LimitState:
+    """Keeps track of how to limit each query
+    Before the first query, limiting_conditions = None, so the first query
+    will apply an ORDER BY ... LIMIT.
+    After that, the groups returned by the first query will be added as a
+    WHERE clause to any secondary query.
+    """
+
+    def __init__(self) -> None:
+        self._initialized = False
+        self.skip_columns: Set[Column] = set()  # Will not be added to limiting conditions
+
+    def update(self, groupby: Collection[Column], snuba_rows: _SnubaData) -> None:
+        if self._initialized:
+            return
+
+        # Only "totals" queries may set the limiting conditions:
+        assert Column(TS_COL_GROUP) not in groupby
+
+        self._groupby = groupby = list(groupby)  # Make sure groupby has a fixed order
```

Comment:
What's the point of defining `self._groupby` and `groupby` ? They are both pointing to the same object anyways